drivers/gpio/gpio-usbio.c

```diff
@@ -127,9 +128,17 @@ static int usbio_gpio_read(struct usbio_gpio_dev *usbio_gpio, u8 gpio_id)
 	packet->bankid = gpio_id / GPIO_PER_BANK;
 	packet->pincount = 1;
 	packet->pins = gpio_id % GPIO_PER_BANK;
-	ret = usbio_transfer(usbio_gpio->pdev, GPIO_READ, packet,
```

Comment:
this is an bad hack, needs to be handled by FW.

Reply:
this is due to the 4-bytes of trailing ZEROs in the gpio payload during the USB control transfer the the FW.
Do we have a better to address this issue if the FW already uses protocol?